### DISCUSSION TITLE
Please don't use Cloudflare

### DISCUSSION THREAD

**Discussion by ureru** [stoatchat/discussions#248] [Feature Suggestions]
This is a message to everyone who thinks Revolt is private and secure. Revolt uses Cloudflare just like Discord does. Cloudflare is a man in the middle that has access to everything sent to and from the site in cleartext. You're not going to get proof that this isn't being used to spy on people presumably. No one should be using it, nor the other services that do the same thing. End to end encryption may help, but the keys to decrypting the messages need to be, at minimum, not even something Revolt has ANY access to before that will make any difference at all and even then it's not good at all to hand encrypted data to a ton of unknown people unnecessarily (you don't just use e2ee to try and fix all of your problems by itself while also using things like cloudflare, not decentralizing, etc.), and it would also have to be implemented for entire servers and channels including voice chat instead of only for DMs and it would also have to be enabled by default for everything, and supposedly we won't get e2ee until v0.6.0 anyway. If you need an anti-DDOS solution, it seems that there should be much easier solutions to at least heavily mitigate it and I doubt cloudflare does much better than those solutions would statistically.

**Comment by insertish** [comment]
So what do you suggest using?

It's also worth pointing out that:
- as I'm still a student and there's no sort of, super reliable income, I don't think I can afford to pay for anything
- Cloudflare has already helped me block several attacks, especially very targeted API attacks, I can actually completely mitigate any sort of weird application level attack with relative ease

**Reply by BitForger** [reply]
If you're trying to block DDoS you should rate-limit your API's and use a circuit-breaker to put them in a cool-down list that effectively black-holes the requests. This can be done at the API level but at some point you may need to do it at higher level in the network before the requests hit your API. It's hard to know exactly what can be done because I'm not intimately familiar with your infrastructure but there are solutions available that don't require cloudflare and may not necessarily incur costs

**Reply by Rexogamer** [reply]
(note: rate limiting is planned (see #322 for more info))

**Reply by ggtylerr** [reply]
Old post but I wanted to chime in on this.

Hetzner provides their own DDOS protection, free of charge, though I can't say how effective it is since I don't use the service. 
You can use ModSecurity ( [nginx](https://github.com/SpiderLabs/ModSecurity-nginx) / [Apache](https://github.com/SpiderLabs/ModSecurity/tree/v2/master) ) or any other WAF. There's also [this list](https://framagit.org/dCF/deCloudflare/-/raw/master/tool/useful_ip_list/vps.txt) with VPS IPs that can be blocked.
Of course, there's rate limiting as well, but that's already implemented.

All of these are free solutions (unless you use a paid WAF) and don't require any major infrastructure changes.

**Reply by Kellegram** [reply]
Unlikely that alternatives have an infrastructure as powerful as Cloudflare's for DDoS protection. Cloudflare is good for what they provide. The paranoia of the main post is frankly pointless as if using Cloudflare is the problem, they better start building a shack in the woods, coz there are far bigger problems here x)

**Reply by ggtylerr** [reply]
> Unlikely that alternatives have an infrastructure as powerful as Cloudflare's for DDoS protection. Cloudflare is good for what they provide. The paranoia of the main post is frankly pointless as if using Cloudflare is the problem, they better start building a shack in the woods, coz there are far bigger problems here x)

Okay, granted, OP's text reads like its a bit paranoid. But there is good reason.

The ["deCloudflare"](https://framagit.org/dCF/deCloudflare/-/blob/master/readme/en.md) project has been around for a while, and is constantly removed on most platforms (latest one in my memory is [Codeberg.](https://codeberg.org/crimeflare/cloudflare-tor)) You can read the text yourself, but really, OP is scratching the surface.

Of course, you guys can choose to continue using it, but for a "Privacy First" platform, it really seems like a kick in the gut to do so, and is downright deceptive for some people.

**Reply by Kellegram** [reply]
Wow, this conspiracy paper is painful to read. Most of the things seen as problems are expected and even desired behaviour. If you are worried about Cloudflare, don't use internet, because you are actively using far worse things than CF. Paranoia fueled by lack of knowledge in networking and in general how internet works. ""Privacy"" is a meme, most people who promote privacy-first alternatives don't really know much about networking or the global survaillence, just read too much on some obscure forums and listened to much too some meme tubers and whoever else. I hope Revolt will get rid of the privacy branding, because anything on the internet that cites privacy is misleading at best. Revolt is simply a great and open alternative to Discord, that's all it is. If you think you are using something on the internet and have any privacy, you are oh so wrong. Even Tor nowadays is not really all that anonymous if you are a target. And if you aren't a target you have no reason to act so defensive online. If anything, by trying to be """private""" with many quotation marks, you are more likely to become a target due to suspicious behaviour. Reduce your online fingerprint, that's all you can do. Conspiracy theories aren't helping anyone, things like what's in your link is either mistakenly or purposefully written in a very misleading way, flipping positives of Cloudflare and pointing out edge cases and disadvantages, which for most do NOT outweigh the advantages. 

If you want negative criticism to be taken seriously, you have to present the facts and acknowledge positives, otherwise it just looks like yet another 4chan meme.

**Reply by moeenio** [reply]
@Kellegram what do you mean by "privacy" exactly here? Why can I not have *any* "privacy" when using a service which collects the least possible data and encrypts what it does collect? And what could I be using that would be "worse than CF"? Please educate me.

**Reply by JacksonUptain** [reply]
Cloudflare does work. Pls use it, its the best

**Comment by UnixPhonez** [comment]
I agree cloudflare is terrible for privacy and would like to see this changed in the future, but I also understand as a student the income scale is low. But cloudflare should be changed once revolt gets an income it's simple as that.

**Comment by FelicityIris** [comment]
I second to replacing Cloudfare with a better service. The website mentioned earlier has a page with free alternatives and methods to Cloudfare if needed. Scroll to the Web Developer Section on [this page](https://0xacab.org/dCF/deCloudflare/-/blob/master/readme/en.action.md#Looking_for_hosting_service) or just visit [this page](https://0xacab.org/dCF/deCloudflare/-/blob/master/subfiles/alternative/domaindns.md) or [this page](https://0xacab.org/dCF/deCloudflare/-/blob/master/subfiles/alternative/cloudflare.md) for a direct list.
On top of privacy, Cloudfare is known to have [wide ethics issues](https://0xacab.org/dCF/deCloudflare/-/blob/master/readme/en.ethics.md) and has a history of supporting White Supremacist Extremist Websites so here is another incentive on why not to use Cloudfare. I personally would not hold my trust for such a service.

**Comment by yani** [comment]
A federation would actually make this good. 

Otherwise it's just a discord clone with the same promises as them. And an opensource client.
